### PR TITLE
Community Convective Cloud (C3) scheme

### PR DESCRIPTION
**Pull request by haiqinli**
## Description

1. Name the new unified convective parameterization as "Community Convective Cloud (c3) parameterization";
2. Update the aerosol-aware for GF;
3. Merge the aerosol-aware and RRFS configuration from GF into C3;
4. Make the closures as input namelist options for C3 and GF.

## Testing

How were these changes tested?  Yes
What compilers / HPCs was it tested with?  Intel compiler on Hera
Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)  
A new regression test run case of C3 convection should be added.
Have the ufs-weather-model regression test been run? On what platform?  Yes, on Hera with Intel compiler.
- Will the code updates change regression test baseline? If yes, why? Yes, the GF code was updated.

## Dependencies
- depends on https://github.com/ufs-community/ccpp-physics/pull/73
Do PRs in upstream repositories need to be merged first? Yes.

### CODE REVIEW

**Comment by lisa-bengtsson** [comment]
@haiqinli I wonder if we should create a suite def file also for the RRFS configuration with C3? I made 2 for the global only (GFSv17 and GFSv17 coupled)

**Comment by haiqinli** [comment]
@lisa-bengtsson Yes, I will add a HRRR_c3 suite for RRFS when I update this PR.

**Comment by BrianCurtis-NOAA** [comment]
This needs to be reviewed before it can get into UFSWM commit queue.

**Review by DusanJovic-NOAA — APPROVED** [review]

File: ccpp/data/GFS_typedefs.F90 [hunk 1]
```diff
@@ -1160,13 +1160,16 @@ module GFS_typedefs
...
-    integer              :: imfdeepcnv_unified  = 5 !< flag for the unified convection scheme
+    integer              :: imfdeepcnv_c3       = 5 !< flag for the Community Convective Cloud (C3) scheme
     logical              :: hwrf_samfshal           !< flag for HWRF SAMF shalcnv scheme (HWRF)
     integer              :: isatmedmf       !< flag for scale-aware TKE-based moist edmf scheme
                                             !<     0: initial version of satmedmf (Nov. 2018)
                                             !<     1: updated version of satmedmf (as of May 2019)
     integer              :: isatmedmf_vdif  = 0 !< flag for initial version of satmedmf (Nov. 2018)
     integer              :: isatmedmf_vdifq = 1 !< flag for updated version of satmedmf (as of May 2019)
+    integer              :: ichoice         = 0 !< flag for closure of C3/GF deep convection
+    integer              :: ichoicem        = 13!< flag for closure of C3/GF mid convection
```
Comment (BrianCurtis-NOAA) [inline ccpp/data/GFS_typedefs.F90:1171]: I'm just curious, the naming for this and the following variable are inconsistent, but is there a purpose that one has _s and the other without an underscore?

**Review by BrianCurtis-NOAA — APPROVED** [review]

File: ccpp/data/GFS_typedefs.F90 (hunk 1 above)
Comment (haiqinli) [reply ccpp/data/GFS_typedefs.F90:1171]: @BrianCurtis-NOAA The name of ichoice_s for shallow convection closure follows the GF convection in cu_gf_driver.F90. Thanks.

**Comment by DeniseWorthen** [comment]
@haiqinli @BrianCurtis-NOAA This PR also appears to depend on a CCPP changes but no information is provided here or at the UFS PR. Is there a CCPP PR associated with this ?

**Comment by haiqinli** [comment]
@DeniseWorthen Yes, the associated CCPP PR is https://github.com/ufs-community/ccpp-physics/pull/73. Thanks.

**Comment by DeniseWorthen** [comment]
@haiqinli Please link that PR to this PR. Do that by editing your PR description with "- depends on" and the link to the CCPP PR.  

In the above PR information you answered "no" to the question of whether this PR has dependencies.

**Comment by haiqinli** [comment]
@DeniseWorthen The PR description has been updated accordingly. Thanks.

**Comment by grantfirl** [comment]
@haiqinli The physics PR was merged. Please revert .gitmodules and update the ccpp/physics submodule pointer in preparation for the fv3atm merge.

**Comment by jkbk2004** [comment]
@haiqinli ccpp pr was merged. Can you update ccpp/physics submodule pointer and revert change in gitmodules? Correct hash is https://github.com/ufs-community/ccpp-physics/commit/427c77ba16f62bd84996f91b09655fff53abe549

**Comment by jkbk2004** [comment]
@haiqinli can you resolve the conversation as well?

**Comment by haiqinli** [comment]
@jkbk2004 @grantfirl Yes, I am working on it, and will update when it is done.

**Comment by haiqinli** [comment]
I have some difficulties to do git push right now, and will continue to try it.

[Haiqin.Li@hfe04 fv3atm]$ git push https://github.com/haiqinli/fv3atm.git
fatal: unable to access 'https://github.com/haiqinli/fv3atm.git/': Could not resolve host: github.com; Unknown error

**Comment by haiqinli** [comment]
@jkbk2004 @grantfirl Revert submodule pointer to ufs-community/ccpp-physics@427c77b is done. Thanks.

**Pull request merged**